### ISSUE TITLE
Add capability to display all types of warnings handled and their definitions (security for dummies)

### ISSUE THREAD

**Issue by antoine-coulon** [NodeSecure/ci#5]
Initially suggested by @RomainLanz

We are introducing a way to configure specifically `warnings`  through a `.nodesecurerc` file (#3). 

However, those warnings meanings and consequences are not easy to understand though. 
@fraxken already initiated the work and added a _Warnings Legends_ in the [README.md of the @nodesecure/js-x-ray package](https://github.com/NodeSecure/js-x-ray).

#### How to retrieve these legends ?

At the beginning, we could imagine a static data structure to include these __Warnings Legends__ in the library. However this does not scale well, as it could introduce differences when adding more warnings to ` js-x-ray`.

#### How to display these legends ?

The goal would be to access those legends directly from `@nodesecure/ci` using either:
- the CLI, by using a custom command printing the legends. **table** dependency could be used (waiting for #3)
- via a static web page referencing these legends and in-depth ressources. The web page could be hosted via a local server.

**Comment by fraxken** [comment]
On our last meeting we discussed the choice to probably manage this within the i18n project to include all translations and informations in tools.

Maybe a good way of solving the problem is to:
- Complete JS-X-Ray to return the list of warnings with an i18n code in each warnings.
- Match the code in i18n to get the right translation.

What do you think?

**Comment by antoine-coulon** [comment]
> On our last meeting we discussed the choice to probably manage this within the i18n project to include all translations and informations in tools.
> 
> Maybe a good way of solving the problem is to:
> 
> * Complete JS-X-Ray to return the list of warnings with an i18n code in each warnings.
> * Match the code in i18n to get the right translation.
> 
> What do you think?

Yes you're absolutely right. We could probably open those issues in [JS-X-Ray](https://github.com/NodeSecure/js-x-ray/issues) and [i18n](https://github.com/NodeSecure/i18n/issues) and then the current issue here would be the final integration of both (by probably printing these translated warnings in the CLI via a command).

**Comment by tony-go** [comment]
Also agree with @fraxken POV ^^

**Comment by Mathieuka** [comment]
I take the issue

**Comment by fraxken** [comment]
`v1.3.0` of i18n and `v4.4.0` of js-x-ray implement required API

Example taken from JS-X-Ray documentation:
```js
import * as jsxray from "@nodesecure/js-x-ray";
import * as i18n from "@nodesecure/i18n";

console.log(i18n.getToken(jsxray.warnings.parsingError.i18n));
```

**Comment by fraxken** [comment]
Fetching warnings i18n description with JS-X-Ray 6.x and i18n 3.2.0

```js
import * as jsxray from "@nodesecure/js-x-ray";
import * as i18n from "@nodesecure/i18n";

console.log(i18n.getTokenSync(jsxray.warnings["obfuscated-code"].i18n));
```

**Comment by antoine-coulon** [comment]
Done by @halcin in #92, will be released in v1.7.0

@all-contributors please add @halcin  for code
